drivers/gpu/drm/msm/adreno/a5xx_gpu.c

```diff
@@ -1770,7 +1770,7 @@ struct msm_gpu *a5xx_gpu_init(struct drm_device *dev)
 
 	nr_rings = 4;
 
-	if (config->info->revn == 510)
+	if (config->info->revn == 510 || config->info->revn == 512)
```

Comment:
the same likely can be done for 508 & 509, probably (for devices based on sdm630 and sdm636) ?

Reply:
I do not think we have devices with sdm630/a508 to test, but @M0Rf30 I think has tulip with a509 ?

Reply:
As far as i know a5xx preemption is broken globaly and there are parts what are not implemented in mesa so i could disable for every a5xx with remove the if.